client/state/plugins/marketplace/actions.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * External dependencies
+ */
+import type { Action } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import 'calypso/state/plugins/init';
+
+import { PLUGINS_MARKETPLACE_PRIMARY_DOMAIN_UPDATE } from 'calypso/state/action-types';
+
+interface ISetPrimaryDomainAction extends Action {
```

Comment:
Have you considered moving types to a separate file that contains types only?

Reply:
I think having the type shape here is quite useful for anyone who wants to use this action. Otherwise they have to go through multiple code files to understand the big picture.

Reply:
On the contrary, my opinion is that having types at the same place makes more sense for 3 reasons:

* Interconnected types in separate files are like puzzle pieces; to understand the whole picture you need to solve the puzzle first.
* Having the types at the same place allows the reader of the code to better understand what we're typing and it's also more predictable as to where to import them from (and where to look for them).
* Gathering the types of a certain group in a `types.ts` file alongside the files that consume them is already common in Calypso.

To me creating the types all over the place creates a small mess 😉 But that's just my $0.02. What do you think about all this?

Reply:
Actually I write code in a reducer driven manner and normally I maintain the state shape near the reducer. I guess this is a personal pattern I am familiar with. But if having a types.ts file is a more familiar practice in calypso it is definitely better to adopt that practice since this can be argued both ways :)

Reply:
As the author of the PR, you get to choose. This is my personal opinion and not a blocker for merging this PR by any means 😉